object_storage_access_manager/osam/tasks.py

```
@@ -280,26 +281,38 @@ def delete_obs_user_account_if_not_used_by_keycloak_account(
             )
 
 
-def get_user_s3_credentials(user: str):
+def get_user_s3_credentials(user: str) -> dict:
     """
     Retrieves the S3 access and secret keys for a given user.
 
     Args:
         user (str): The username for whom to retrieve S3 credentials.
 
     Returns:
-        dict: A dictionary containing the 'access_key' and 'secret_key' for the user's S3 storage.
+        dict: A dictionary containing 'access_key', 'secret_key', 'endpoint_url', 'region_name'
+        for the user's S3 storage.
     """
     try:
         obs_user = get_keycloak_handler().get_obs_user_from_keycloak_username(user)
-        if obs_user:
-            if access_key := get_ovh_handler().get_user_s3_access_key(obs_user):
-                secret_key = get_ovh_handler().get_user_s3_secret_key(obs_user, access_key)
-                return {"access_key": access_key, "secret_key": secret_key}
-            return {"detail": f"Error reading user {user} from OVH."}
+
+        if not obs_user:
+            raise RuntimeError(f"No s3 credentials associated with {user}")
```

Just let the error propagate instead of returning
```py
return {"detail": f"No s3 credentials associated with {user}"}
```
because you don't want the caller to test every possible error case.